### PR TITLE
[Merged by Bors] - Changed Msaa to Enum

### LINKED ISSUE #6931: Setting MSAA to 1 sample to disable it is really confusing
## What problem does this solve or what need does it fill?

It's confusing that you have to set the `samples` value to 1 to turn off MSAA.

## What solution would you like?

Ira presented a pretty good one: https://discord.com/channels/691052431525675048/1051819150189809715/1051822325907079208 Just have the MSAA be a field on the camera like HDR.

## What alternative(s) have you considered?

Could have an MSAA enum like:

```rust
enum MSAA {
    On(u32),
    Off,
}
```

and insert that as a resource.

### PR DESCRIPTION
**Pull request by Sjael**
# Objective

Fixes #6931 

Continues #6954 by squashing `Msaa` to a flat enum

Helps out  #7215 

# Solution
```
pub enum Msaa {
    Off = 1,
    #[default]
    Sample4 = 4,
}
```

# Changelog

- Modified
    - `Msaa` is now enum
    - Defaults to 4 samples
    - Uses `.samples()` method to get the sample number as `u32`

# Migration Guide
```
let multi = Msaa { samples: 4 } 
// is now
let multi = Msaa::Sample4

multi.samples
// is now
multi.samples()
```

### CODE REVIEW

File: examples/3d/fxaa.rs [hunk 1]
```diff
@@ -13,8 +13,8 @@ use bevy::{
 
 fn main() {
     App::new()
-        // Disable MSAA be default
-        .insert_resource(Msaa { samples: 1 })
+        // Disable MSAA by default
+        .insert_resource(Msaa::default())
```
Comment (nicopap) [inline examples/3d/fxaa.rs:17]:
```suggestion
        .insert_resource(Msaa::Off)
```
This doesn't match the previous value

**Review by nicopap — CHANGES_REQUESTED** [review]
LGTM, just a minor change.

File: crates/bevy_render/src/view/mod.rs [hunk 2]
```diff
@@ -61,25 +61,33 @@ impl Plugin for ViewPlugin {
...
+        *self as u32
+    }
 }
 
 impl Default for Msaa {
     fn default() -> Self {
-        Self { samples: 4 }
+        Self::Sample4
     }
 }
```
Comment (Aceeri) [inline crates/bevy_render/src/view/mod.rs:92]: Could make this a `#[derive(Default)]`

**Review by Aceeri — CHANGES_REQUESTED** [review]

File: examples/3d/fxaa.rs (hunk 1 above)
Comment (Sjael) [reply examples/3d/fxaa.rs]: Woops, thanks

**Comment by wilk10** [comment]
Thanks a lot for this! I'm very much in favour of this change, it's much clearer.

This is a breaking change, can you please re-add the Changelog and Migration Guide sections?

**Comment by nicopap** [comment]
There is a weird file called `q` that has 52K lines that was added to the merge set, could you remove it?

**Review by nicopap — APPROVED** [review]
LGTM

**Review by Aceeri — APPROVED** [review]

**Comment by Sjael** [comment]
> There is a file with the entire bevy git history called `q` that has 52K lines that was added to the merge set, could you remove it?

Yeah, I accidentally saved my git log when trying to wq lol

**Review by wilk10 — APPROVED** [review]
Thanks a lot! LGTM

**Review by alice-i-cecile — APPROVED** [review]
Nice and clean.

bors r+

**Pull request closed without merging**